src/openfermion/utils/_diagonal_coulomb_trotter_error.py

```diff
@@ -0,0 +1,290 @@
+        modes_acted_on_by_potential_term = set()
+
+        for potential_term_action in potential_term.terms:
+            modes_acted_on_by_potential_term.update(
+                set(operator[0] for operator in potential_term_action))
+
+        if not modes_acted_on_by_potential_term:
+            continue
+
+        potential_term_mode_mask = numpy.logical_or.reduce(
+            [kinetic_mask[mode] for mode in modes_acted_on_by_potential_term])
+
+        for kinetic_term in kinetic_terms[potential_term_mode_mask]:
+            inner_commutator_term = (
+                commutator_ordered_diagonal_coulomb_with_two_body_operator(
+                    potential_term, kinetic_term))
+
+            modes_acted_on_by_inner_commutator = set()
+            for inner_commutator_action in inner_commutator_term.terms:
+                modes_acted_on_by_inner_commutator.update(
+                    set(operator[0] for operator in inner_commutator_action))
+
+            if not modes_acted_on_by_inner_commutator:
+                continue
+
+            inner_commutator_mode_mask = numpy.logical_or.reduce(
+                [potential_mask[mode]
+                 for mode in modes_acted_on_by_inner_commutator])
+
+            # halved_potential_terms for T+V order, potential_terms for V+T
+            for outer_potential_term in outer_potential_terms[
+                    inner_commutator_mode_mask]:
+                commutator_ordered_diagonal_coulomb_with_two_body_operator(
+                    outer_potential_term, inner_commutator_term,
+                    prior_terms=error_operator)
+
+            inner_commutator_mode_mask = numpy.logical_or.reduce(
+                [kinetic_mask[qubit]
+                 for qubit in modes_acted_on_by_inner_commutator])
+
+            # kinetic_terms for T+V order, halved_kinetic_terms for V+T
+            for outer_kinetic_term in outer_kinetic_terms[
+                    inner_commutator_mode_mask]:
+                commutator_ordered_diagonal_coulomb_with_two_body_operator(
+                    outer_kinetic_term, inner_commutator_term,
+                    prior_terms=error_operator)
+
+    # Divide by 12 to match the error operator definition.
+    # If order='V+T', also flip the sign to account for inner_commutator_term
+    # not flipping between the different orderings.
+    if order == 'T+V':
+        error_operator /= 12.0
+    else:
+        error_operator /= -12.0
+
+    return error_operator
+
+
+def fermionic_swap_trotter_error_operator_diagonal_two_body(hamiltonian):
+    """Compute the fermionic swap network Trotter error of a diagonal
+    two-body Hamiltonian.
+
+    Args:
+        hamiltonian (FermionOperator): The diagonal Coulomb Hamiltonian to
+                                       compute the Trotter error for.
+
+    Returns:
+        error_operator: The second-order Trotter error operator.
+
+    Notes:
+        Follows Equation 9 of Poulin et al.'s work in "The Trotter Step
```

Comment:
Just give the arXiv references instead of listing the authors and name of paper.

Reply:
Done.